lib/loader/loader.c

```diff
@@ -238,3 +231,90 @@ struct evmc_instance* evmc_load_and_create(const char* filename,
 
     return instance;
 }
+
+struct evmc_instance* evmc_load_and_configure(const char* config,
+                                              enum evmc_loader_error_code* error_code)
+{
+    enum evmc_loader_error_code ec = EVMC_LOADER_SUCCESS;
+    struct evmc_instance* instance = NULL;
+
+    char config_copy_buffer[PATH_MAX_LENGTH];
+    if (strcpy_sx(config_copy_buffer, sizeof(config_copy_buffer), config) != 0)
+    {
+        ec = set_error(EVMC_LOADER_INVALID_ARGUMENT,
+                       "invalid argument: configuration is too long (maximum allowed length is %d)",
+                       (int)sizeof(config_copy_buffer));
+        goto exit;
+    }
+
+    char* options = "";
+    char* path_sep = strchr(config_copy_buffer, ',');
+    if (path_sep)
+    {
+        *path_sep = '\0';
+        options = path_sep + 1;
+    }
+    const char* path = config_copy_buffer;
+
+    instance = evmc_load_and_create(path, error_code);
+    if (!instance)
+        return NULL;
+
+    if (instance->set_option == NULL && strlen(options) != 0)
+    {
+        ec = set_error(EVMC_LOADER_INVALID_OPTION_NAME, "%s (%s) does not support any options",
+                       instance->name, path);
+        goto exit;
+    }
+
+
+    while (strlen(options) != 0)  // TODO: Can be optimized as *options != 0 because MSVC sucks.
```

Comment:
Can or Can't?

Reply:
Can. The `strlen(options) != 0` is not optimized by MSVC, but is more readable.